src/option.c

```diff
@@ -4952,6 +5010,10 @@ set_num_option(
     need_mouse_correct = TRUE;
 #endif
 
+    // May set global value for local option.
+    if ((opt_flags & (OPT_LOCAL | OPT_GLOBAL)) == 0)
+	*(long *)get_varp_scope(&(options[opt_idx]), OPT_GLOBAL) = value;
```

Comment:
:memo: Boolean or string options override the global value before checking errors. Now number options do the same.